cpp/src/arrow/parquet/parquet-io-test.cc

```diff
@@ -52,25 +53,104 @@ template <typename TestType>
 struct test_traits {};
 
 template <>
+struct test_traits<BooleanType> {
+  static constexpr ParquetType::type parquet_enum = ParquetType::BOOLEAN;
+  static constexpr LogicalType::type logical_enum = LogicalType::NONE;
+  static uint8_t const value;
+};
+
+const uint8_t test_traits<BooleanType>::value(1);
+
+template <>
+struct test_traits<UInt8Type> {
+  static constexpr ParquetType::type parquet_enum = ParquetType::INT32;
+  static constexpr LogicalType::type logical_enum = LogicalType::UINT_8;
```

Comment:
Yeah, I believe so. The problem is, at the moment Parquet 2 is not "in production". Even parquet-mr is still writing Parquet 1 files by default, so we should also default to parquet 1.x behavior for the time being